packages/perseus/src/widgets/interactive-graphs/math/geometry.test.ts

```diff
@@ -0,0 +1,178 @@
+import {segmentsIntersect} from "./geometry";
+
+import type {Segment} from "./geometry";
+
+describe("segmentsIntersect", () => {
+    it("returns false when segments have zero length", () => {
+        const segment1: Segment = [
+            [0, 0],
+            [0, 0],
+        ];
+        const segment2: Segment = [
+            [1, 1],
+            [1, 1],
+        ];
+        expect(segmentsIntersect(segment1, segment2)).toBe(false);
+    });
+
+    it("returns false when segments are the same", () => {
+        // TODO(benchristel): is this desired behavior?
+        const segment1: Segment = [
+            [0, 0],
+            [1, 1],
+        ];
+        const segment2: Segment = [
+            [0, 0],
+            [1, 1],
+        ];
+        expect(segmentsIntersect(segment1, segment2)).toBe(false);
+    });
+
+    it("returns false when an endpoint touches the other segment (lambda = 1)", () => {
```

Comment:
This behaviour smells of a rounding issue. :) It feels like a segment ending somewhere on the other segment should intersect (given the "rules" from my earlier comment.  Probably something to verify with the Math content folks.

Reply:
Changing the check from `lambda < 1` to `lambda <= 1` makes the function return `true` in this case. Rounding could also be an issue but my tests don't check for that.

[According to Wikipedia](https://en.wikipedia.org/wiki/Line_segment), line segments can be closed (including both endpoints), open (excluding both endpoints) or half-open. So any combination of `<=` or `<` checks might be mathematically valid depending on what type of segment we're considering. I think this is all worth documenting.